Improvements

evandrocoan opened the pull request:
I think I worked around every big issue I could find.

evandrocoan committed:
Merged all keybind files to Default.sublime-keymap as they are all

equal.

evandrocoan committed:
Migrated the changes from User/Preferences to the Default package.

evandrocoan committed:
Merge remote-tracking branch 'rctay/master'

# Conflicts:
#	Preferences.sublime-settings

evandrocoan committed:
Documented the settings file Preferences.sublime-settings

evandrocoan committed:
Fixed the packages not replacing the default Sublime Text keys

Thread:
Is possible to package override a Sublime Text Default keybinding?
https://forum.sublimetext.com/t/is-possible-to-package-override-a-sublime-text-default-keybinding/31688

evandrocoan committed:
Forced the sublime text build panel closing repeatedly because

opens itself up to 1 second after the build has started.

evandrocoan committed:
Set to copy the view syntax settings on pipe_views.py

evandrocoan committed:
Set to also remember the caret position alongside with the view

port position.

evandrocoan committed:
Merge remote-tracking branch 'rctay/master'

# Conflicts:
#	Preferences.sublime-settings

evandrocoan committed:
Set to pipe_views.py also copy the regex file setting to allow the

mouse double click go to feature.

evandrocoan committed:
Set buildview.enabled as false by default

Preferences.sublime-settings

evandrocoan committed:
Increased commands.py build panel hiding close up to 2 seconds.

evandrocoan committed:
Created a .no-sublime-package for all packages, in attempt to fix

evandrocoan/ITE 108 - Expand Region keybind is not working on stable version

evandrocoan committed:
Fix the `Build output` tab is not reused after project change or

Sublime Text Restart. Fixes rctay/sublime-text-2-buildview#23 - The
`Build output` tab is not reused after project change or Sublime
Text Restart

evandrocoan committed:
Created the setting "buildview.focus_build": false, to control

whether automatically change your current view to the `Build
output` creating a new build.

evandrocoan committed:
Fixed the Build output view opening when you pressed Ctrl+Shift+B,

by deprecating the Default *.sublime-keymap files with the
on_window_command() event listener.

evandrocoan committed:
Removed outdated comment on commands.py

evandrocoan committed:
Fixed the build view not working if the view is moved to a new

window and fixed it not saving the scroll positon when the view is
closed.

rctay commented on commands.py:
```diff
@@ -133,24 +145,13 @@ def on_query_context(self, view, key, *args):
         # [2] https://forum.sublimetext.com/t/output-panel-hotkey/15628/2
-        def hide_panel():
-            window.run_command("hide_panel")
-        sublime.set_timeout(hide_panel, 1)
```
Any reason why hiding of the Build panel is dropped?

evandrocoan replied:
Because you can explicitly disable the build panel with `show_panel_on_build`. We could say that when someone explicitly enables this package with `"buildview.enabled": true`, they should disable the build panel with `"show_panel_on_build": false`.

Due the bug:
1. https://github.com/SublimeTextIssues/Core/issues/679 show_panel_on_build setting isn't loaded from syntax specific settings

Sublime Text is not loading the setting `show_panel_on_build` from the user projects settings. Then, if someone enables this package per project only, they will not be able to stop the build panel from showing up. However, the fix is very simple, as explained on the issue (https://github.com/SublimeTextIssues/Core/issues/679) and can be performed by any user. For example, my Sublime Text is already patched against that.

I got tired of living with those small glitches/bugs which can be extremely easy fixed by the Sublime Text developers. Then, I decided to stop patching our packages with very ugly and hardly working hacks, and let the affected users directly patch Sublime Text files. Of course, only if they can. Otherwise we would still have to keep our packages with those very ugly hacks.

evandrocoan committed:
Fixed pipe_views.py not detecting the build view it is the only one

evandrocoan committed:
Fixed copy_view_settings() resetting/copying None settings

evandrocoan committed:
Do not search for old build views when there are too much views

because this would probably hand Sublime Text

evandrocoan committed:
Revert "Created a .no-sublime-package for all packages, in attempt to fix"

This reverts commit aadb0ee9bef21ab76d31d155653be461a58f07ad.

evandrocoan committed:
Added support to new settings from default package

rctay commented:
Thank you @evandrocoan.

>

evandrocoan committed:
Updated README.md installation instructions

rctay commented:
Hi @evandrocoan would it be OK for you to rebase your branch on top of this repo's master?

evandrocoan commented:
I think I did not understand correctly.

There are few ways to integrated my branch on your master:
1. You can merge this pull request squashing (all my commits would become a single).
1. You can merge this pull request without squashing (all my commits are kept).
1. You can rebase this pull request and no merge commit would be created and it would be like I was committing directly to your master.
1. You can cherry pick only the commits you would like, directly into your master
1. You can cherry pick only the commits you would like, into some other branch and either merge it, squash it or rebase it into your master.

For me, you can merge this any ways you would like.

evandrocoan committed:
Fixed README.md installation instructions

evandrocoan committed:
Make the README.md instructions clear

evandrocoan committed:
Fixed markdown list not rendering on packagecontorl.io engine

evandrocoan committed:
Fixed packagecontrol.io not interpreting the markdown lists

continuations.

evandrocoan committed:
Created the settings.json with Sublime Text version 3999 marking

the head commit today as the last supported version of Sublime
Text 3. This is used to start developing the migration for Sublime
Text 4. New updates to Sublime Text 3 should no happen. If required, a tag 3998 should be created inside settings.json with
the changes required for this new version.

evandrocoan committed:
Fixed ImportError: No module named 'pipe_views'

Traceback (most recent call last):
  File "F:\SublimeText\Lib\python33\sublime_plugin.py", line 308, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "F:\SublimeText\Data\Packages\BuildView\commands.py", line 7, in <module>
    from pipe_views import PipeViews, proxy_settings
ImportError: No module named 'pipe_views'

evandrocoan committed:
Merge remote-tracking branch 'rctay/master'

# Conflicts:
#	README.md
#	commands.py
#	pipe_views.py
